shopfloor/services/cluster_picking.py

```diff
@@ -501,6 +523,34 @@ def _scan_line_by_product(self, picking, move_line, product):
                 move_line,
                 message=self.msg_store.product_multiple_packages_scan_package(),
             )
+        self._set_or_increment_qty_done(move_line, 1.0)
+        return self._response_for_scan_destination(move_line)
+
+    def _scan_line_by_packaging(self, picking, move_line, packaging):
+        """Packaging scanned, check if we can work with it.
+
+        If the packaging related product is part of several packages in the same location,
+        we can't be sure it's the correct one, in such case, ask to scan a package
+        """
+        product = packaging.product_id
+        if move_line.product_id.tracking in ("lot", "serial"):
+            return self._response_for_start_line(
+                move_line, message=self.msg_store.scan_lot_on_product_tracked_by_lot()
+            )
+        other_product_lines = picking.move_line_ids.filtered(
+            lambda l: l.product_id == product and l.location_id == move_line.location_id
+        )
+        packages = other_product_lines.mapped("package_id")
+        # Do not use mapped here: we want to see if we have more than one package,
+        # but also if we have one product as a package and the same product as
+        # a unit in another line. In both cases, we want the user to scan the
+        # package.
+        if packages and len({line.package_id for line in other_product_lines}) > 1:
+            return self._response_for_start_line(
+                move_line,
+                message=self.msg_store.product_multiple_packages_scan_package(),
+            )
+        self._set_or_increment_qty_done(move_line, packaging.qty)
```

Comment:
This function will modify the qty_done on the move_line ! This is happening before any work is done by the user. This will lead to errors